### PR TITLE
update to use new agency info in updated endpoint

### LINKED ISSUE #212: [Agency Dashboard] Add agency about page information to the justice counts endpoint
Need text underneath the agency name, the links to agency email and website, the jurisdiction text.

### PR DESCRIPTION
**Pull request by terryttsai**
## Description of the change

Update store to consume the new format of the api and display agency name

NOTE: this requires the new backend change to the api made here: https://github.com/Recidiviz/recidiviz-data/pull/17142 otherwise the app will break

## Type of change

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Closes #212

### Development

**This box MUST be checked by the submitter prior to merging**:
- [ ] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: agency-dashboard/src/Header/HeaderBar.tsx [hunk 1]
```diff
@@ -53,11 +55,20 @@ export const HeaderBar = () => {
...
+          agencyName={agencyDataStore.agency?.name || ""}
+        />
+      )}
       <LogoContainer onClick={() => navigate("/")}>
         <Logo src={logo} alt="" />
       </LogoContainer>
-      <HeaderTitle>Justice Counts + Clackamas County Jail</HeaderTitle>
+      <HeaderTitle>
+        {`Justice Counts${
+          agencyDataStore.agency?.name && ` + ${agencyDataStore.agency?.name}`
```
Comment (lilidworkin) [inline agency-dashboard/src/Header/HeaderBar.tsx:69]: Where in the designs does it say Justice Counts + Agency Name?

File: agency-dashboard/src/stores/AgencyDataStore.ts [hunk 2]
```diff
@@ -19,13 +19,16 @@ import {
...
   Metric,
+  UserAgency,
 } from "@justice-counts/common/types";
 import { isPositiveNumber } from "@justice-counts/common/utils";
 import { makeAutoObservable, runInAction } from "mobx";
 
 import { request } from "../utils/networking";
 
 class AgencyDataStore {
+  agency: UserAgency | undefined;
```
Comment (lilidworkin) [inline agency-dashboard/src/stores/AgencyDataStore.ts:30]: Does this need to be undefined-able? I feel like the backend can guarantee that it should always send this?

**Review by lilidworkin — COMMENTED** [review]
Exciting to see more real data coming in!

File: agency-dashboard/src/stores/AgencyDataStore.ts (hunk 2 above)
Comment (terryttsai) [reply agency-dashboard/src/stores/AgencyDataStore.ts:30]: I have it undefined when I first initialize it, then after fetching it from the backend, then I store the user agency there!

**Review by mxosman — COMMENTED** [review]
Code looks good to me, Terry! I'll leave the final approval to Lili to not overstep the ongoing convo!

File: agency-dashboard/src/Header/HeaderBar.tsx (hunk 1 above)
Comment (lilidworkin) [reply agency-dashboard/src/Header/HeaderBar.tsx:69]: Oh yep thanks!

File: agency-dashboard/src/stores/AgencyDataStore.ts (hunk 2 above)
Comment (lilidworkin) [reply agency-dashboard/src/stores/AgencyDataStore.ts:30]: Got it!

**Review by lilidworkin — APPROVED** [review]
Thanks @terryttsai !

**Pull request merged**